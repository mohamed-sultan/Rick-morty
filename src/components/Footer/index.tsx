import { Link as MuiLink } from "@mui/material";
import { styled } from "@mui/material/styles";
import ColorButton from "../Button";
import { useRouter } from "next/router";
import Link from "next/link";

const Footer = styled("footer")({
  display: "flex",
  justifyContent: "center",
  marginBottom: "3rem",
});
const FooterComponent = () => {
  const router = useRouter();
  const page = Number(router?.query?.page) || 1;
  return (
    <Footer>
      {page > 1 && (
        <MuiLink
          underline="none"
          component={Link}
          href={`/${page - 1}`}
          passHref
        >
          <ColorButton text="Previous Page" />
        </MuiLink>
      )}

      <MuiLink underline="none" component={Link} href={`/${page + 1}`} passHref>
        <ColorButton text="Next Page" />
      </MuiLink>
    </Footer>
  );
};

export default FooterComponent;
