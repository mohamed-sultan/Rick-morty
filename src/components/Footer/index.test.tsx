import * as React from "react";
import TestRenderer from "react-test-renderer";
import Footer from "./index";
jest.mock("next/router", () => ({
  useRouter: jest.fn(),
}));
describe("Footer snapshot testing", () => {
  it("renders correctly", () => {
    const tree = TestRenderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
