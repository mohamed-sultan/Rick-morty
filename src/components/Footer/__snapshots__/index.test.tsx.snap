// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer snapshot testing renders correctly 1`] = `
<footer
  className="css-1ezr195"
>
  <a
    className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1cmksfb-MuiTypography-root-MuiLink-root"
    href="/2"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-qt782h-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Next Page
    </button>
  </a>
</footer>
`;
